File: packages/react-native-worklets/android/src/main/java/com/swmansion/worklets/runloop/AnimationFrameCallback.java
package com.swmansion.worklets.runloop;

import com.facebook.jni.HybridData;
import com.facebook.proguard.annotations.DoNotStrip;

@SuppressWarnings("JavaJniMissingFunction")
public class AnimationFrameCallback {

  @DoNotStrip
  @SuppressWarnings({"FieldCanBeLocal", "unused"})
  private final HybridData mHybridData;

  @DoNotStrip
  private AnimationFrameCallback(HybridData hybridData) {
    mHybridData = hybridData;
  }

  public native void onAnimationFrame(double timestampMs);
}

File: packages/react-native-worklets/android/src/main/java/com/swmansion/worklets/runloop/AnimationFrameCallback.kt
package com.swmansion.worklets.runloop

import com.facebook.jni.HybridData
import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
@Suppress("KotlinJniMissingFunction")
class AnimationFrameCallback
    @DoNotStrip
    private constructor(
        @field:DoNotStrip private val mHybridData: HybridData,
    ) {
        external fun onAnimationFrame(timestampMs: Double)
    }
